cpp-client/build.gradle

```diff
@@ -74,44 +74,47 @@ def testCppClient = Docker.registerDockerTask(project, 'testCppClient') {
     copyOut {
         into layout.buildDirectory.dir('test-results')
     }
+    prefix = '/opt/deephaven'
     dockerfile {
         from('deephaven/cpp-client-base:local-build')
         //
         // Build client library.
         //
-        runCommand('mkdir -p /cpp-client/deephaven/{dhcore,dhclient,examples,tests}')
-        copyFile('deephaven/CMakeLists.txt',  '/cpp-client/deephaven/')
-        copyFile('deephaven/dhcore/',  '/cpp-client/deephaven/dhcore/')
-        copyFile('deephaven/dhclient/', '/cpp-client/deephaven/dhclient/')
-        copyFile('deephaven/examples/', '/cpp-client/deephaven/examples/')
-        copyFile('deephaven/tests/', '/cpp-client/deephaven/tests/')
-        runCommand('''set -eux; \\
-                      mkdir -p /cpp-client/deephaven/build; \\
-                      cd /cpp-client/deephaven/build; \\
-                      . /cpp-client/deps/env.sh; \\
-                      cmake -DCMAKE_INSTALL_PREFIX=/cpp-client/install \\
+        runCommand("mkdir -p ${prefix}/src/deephaven/{dhcore,dhclient,examples,tests}")
+        copyFile('deephaven/CMakeLists.txt',  prefix + '/src/deephaven/')
+        copyFile('deephaven/dhcore/',  prefix + '/src/deephaven/dhcore/')
+        copyFile('deephaven/dhclient/', prefix + '/src/deephaven/dhclient/')
+        copyFile('deephaven/examples/', prefix + '/src/deephaven/examples/')
+        copyFile('deephaven/tests/', prefix + '/src/deephaven/tests/')
+        runCommand("PREFIX=\"" + prefix + "\"\n" +
+                   '''set -eux; \\
+                      mkdir -p ${PREFIX}/src/deephaven/build; \\
+                      cd ${PREFIX}/src/deephaven/build; \\
+                      . ${PREFIX}/env.sh; \\
+                      cmake -DCMAKE_INSTALL_PREFIX=$PREFIX \\
                         -DCMAKE_BUILD_TYPE=Release \\
                         -DBUILD_SHARED_LIBS=ON \\
                         .. ; \\
-                      make -j\$NCPUS install
+                      make -j${NCPUS} install
                    ''')
         //
         // Build tests.
         //
-        runCommand('mkdir -p /cpp-client/install/bin /out')
-        copyFile('cpp-tests-to-junit.sh', '/cpp-client/install/bin')
-        runCommand('''set -eux; \\
-                      cd /cpp-client/deephaven/build/tests; \\
+        runCommand("mkdir -p ${prefix}/bin /out")
+        copyFile('cpp-tests-to-junit.sh', "${prefix}/bin/dhcpp-tests-to-junit.sh')
```

Comment:
Looks like you have a quote mismatch here
``"${prefix}/bin/dhcpp-tests-to-junit.sh'``
starts with double quote, ends with single quote